apps/src/code-studio/components/header/HeaderPopup.jsx

```diff
@@ -0,0 +1,113 @@
+import PropTypes from 'prop-types';
+import React, {Component} from 'react';
+import color from '../../../util/color';
+import progress from '../../progress';
+import MiniView from '../progress/MiniView';
+import i18n from '@cdo/locale';
+
+const styles = {
+  container: {
+    textAlign: 'center',
+    cursor: 'pointer'
+  },
+  caret: {
+    fontSize: 40,
+    color: color.orange,
+    lineHeight: '10px'
+  },
+  more: {
+    fontSize: 10,
+    lineHeight: '10px'
+  }
+};
+
+export default class HeaderPopup extends Component {
+  static propTypes = {
+    scriptName: PropTypes.string,
+    scriptData: PropTypes.object,
+    currentLevelId: PropTypes.string,
+    linesOfCodeText: PropTypes.string,
+    minimal: PropTypes.bool,
+    windowHeight: PropTypes.number
+  };
+
+  state = {
+    open: false
+  };
+
+  shouldComponentUpdate(nextProps, nextState) {
+    return (
+      this.props.windowHeight !== nextProps.windowHeight ||
+      this.props.minimal !== nextProps.minimal ||
+      this.state.open !== nextState.open
+    );
+  }
+
+  handleClickOpen = () => {
+    this.setState({open: true});
+
+    progress.retrieveProgress(
+      this.props.scriptName,
+      this.props.scriptData,
+      this.props.currentLevelId
+    );
+  };
+
+  handleClickClose = () => {
+    this.setState({open: false});
+  };
+
+  handleClickBackground = event => {
+    if (event.target === this.refs.background) {
+      this.setState({open: false});
+    }
+  };
+
+  render() {
+    console.log('HeaderPopup render');
+
+    return (
+      <div style={styles.container}>
+        {!this.state.open && (
+          <div>
+            <div className="header_popup_link" onClick={this.handleClickOpen}>
```

Comment:
Here and elsewhere: For accessibility, avoid clickable divs unless there is a `button` or `a` element that can be used to achieve the same functionality.

Reply:
Great point.  I think this would be really good to address comprehensively in a followup.